introduction_to_applying_machine_learning/deepar_chicago_traffic_violations/deepar_chicago_traffic_violations.ipynb

```diff
@@ -0,0 +1,467 @@
+    "\n",
+    "The dataset contains several columns, we will use the Address, violation date, violations for the forecasting.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "datafile = 'Chicago_Speed_Camera_Violations.csv'\n",
+    "speeding_violation_data_path = '{}/{}/{}.csv'.format(bucket,prefix, datafile)\n",
+    "s3_output_path = '{}/{}/output'.format(bucket, prefix)\n",
+    "s3_data_path = '{}/{}/data'.format(bucket, prefix)\n",
+    "train_data_path = '{}/{}/train/train.json'.format(bucket, prefix)\n",
+    "test_data_path = '{}/{}/test/test.json'.format(bucket, prefix)\n",
+    "\n",
+    "url = 'https://data.cityofchicago.org/api/views/hhkd-xvj4/rows.csv?accessType=DOWNLOAD'\n",
+    "\n",
+    "# get the data from City of Chicago site\n",
+    "r = requests.get(url, allow_redirects=True)\n",
+    "open(datafile, 'wb').write(r.content)\n",
+    "\n",
+    "# store it in S3 object\n",
+    "key = '{}/{}'.format(prefix, datafile)\n",
+    "s3client = boto3.client('s3', region_name=region)\n",
+    "with open(datafile, 'rb') as data:\n",
+    "    s3client.upload_fileobj(data, bucket, key)\n",
+    "\n",
+    "# get the object from S3, and display sample rows/columns\n",
+    "print(key)\n",
+    "print(bucket)\n",
+    "\n",
+    "obj = s3client.get_object(Bucket=bucket, Key=key)\n",
+    "pd.set_option('display.max_columns', 500)     \n",
+    "pd.set_option('display.max_rows', 50)    \n",
+    "df = pd.read_csv(io.BytesIO(obj['Body'].read()), encoding='utf-8')\n",
+    "# print first 10 lines to look at part of the dataset\n",
+    "df[['ADDRESS', 'VIOLATION DATE', 'VIOLATIONS']][0:10]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We convert the violation date from string format to date format, determine the range of violation dates, look at how many unique street address/camaera we have in our dataset.\n",
+    "\n",
+    "The data set contains multiple camera locations and daily violation counts. If we imagine that each camera's daily violation count as one time series, we can use Amazon SageMaker’s DeepAR algorithm to train a model for multiple streets simultaneously, and predict violation for multiple street cameras using the Amazon SageMaker’s DeepAR algorithm.\n",
+    "\n",
+    "As described in [Amazon SageMaker DeepAR input/output interface](https://docs.aws.amazon.com/sagemaker/latest/dg/deepar.html#deepar-inputoutput) section, we will convert the data into array, and fill with 0 violation when data for a given camera on a given date is not available. Using Matplotlib library we display each camera location as a timeseries to visualize the data\n"
```

Comment:
[nitpick]
* "fill with 0 violation" --> "use 0 for the violation count"
* "Matplotlib library" --> "the Matplotlib library,"